RevenueCatUI/CustomerCenter/Views/WrongPlatformView.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  WrongPlatformView.swift
+//
+//
+//  Created by Andrés Boedo on 5/3/24.
+//
+
+import Foundation
+import RevenueCat
+import SwiftUI
+
+@available(iOS 15.0, *)
+struct WrongPlatformView: View {
+
+    @State
+    private var store: Store?
+
+    @Environment(\.openURL)
+    private var openURL
+
+    init() {
+        self._store = State(initialValue: nil)
+    }
+
+    init(store: Store) {
+        self._store = State(initialValue: store)
+    }
+
+    var body: some View {
+        VStack {
+
+            switch store {
+            case .appStore, .macAppStore, .playStore, .amazon:
+                let platformName = humanReadablePlatformName(store: store!)
+
+                Text("Your subscription is being billed through \(platformName).")
+                    .font(.title)
+                    .padding()
+                Text("Go the app settings on \(platformName) to manage your subscription and billing.")
+                    .padding()
+            default:
+                Text("Please contact support to manage your subscription")
+                    .font(.title)
+                    .padding()
+            }
+
+            Spacer()
+
+            Button("Contact support") {
+                Task {
+                    openURL(URLUtilities.createMailURL()!)
+                }
+            }
+            .padding()
+
+        }
+        .onAppear {
+            if store == nil {
+                Task {
+                    if let customerInfo = try? await Purchases.shared.customerInfo(),
+                       let firstEntitlement = customerInfo.entitlements.active.first {
+                        self.store = firstEntitlement.value.store
+                    }
+                }
+            }
+        }
+    }
+
+    private func humanReadablePlatformName(store: Store) -> String {
```

Comment:
This is okay for this PR but we should look into finding a different name here for some of the platforms. Most end users won't know what "Stripe", "RCBilling", "External", or even "Promotional" are.

Reply:
I check for them in https://github.com/RevenueCat/purchases-ios/pull/3919/files#diff-d35a1e787947e08d6465b6c369913eea0f5cae5f1b29428539e35162a4092ec3R33 so we don't display these weird unrecognizable names. But was also wondering if we could find something like "Alternative Store" in case someone calls this function in the future from another spot. Any ideas?

Reply:
I think we should just unify them under a single "Web subscription" hat, as an end user that's all you care about. The link will point you in the right direction.

Well, "Promotional" we could rename to something like "free subscription" or something to allude to the fact that you didn't directly pay for it